pkg/operator/staticpod/internal/atomicdir/swap_linux_test.go

```diff
@@ -0,0 +1,222 @@
+		t.Helper()
+		expectNoError(t, err)
+
+		// Make sure the contents are swapped.
+		dirStateA.CheckDirectoryMatches(t, dirB)
+		dirStateB.CheckDirectoryMatches(t, dirA)
+	}
+
+	testCases := []struct {
+		name        string
+		setup       func(t *testing.T, tmpDir string) (dirA, dirB string)
+		checkResult func(t *testing.T, dirA, dirB string, err error)
+	}{
+		{
+			name: "success with absolute paths",
+			setup: func(t *testing.T, tmpDir string) (string, string) {
+				dirA := filepath.Join(tmpDir, "a")
+				dirB := filepath.Join(tmpDir, "b")
+
+				dirStateA.Write(t, dirA)
+				dirStateB.Write(t, dirB)
+
+				return dirA, dirB
+			},
+			checkResult: checkSuccess,
+		},
+		{
+			name: "success with path A relative",
+			setup: func(t *testing.T, tmpDir string) (string, string) {
+				dirA := filepath.Join(tmpDir, "a")
+				dirB := filepath.Join(tmpDir, "b")
+
+				dirStateA.Write(t, dirA)
+				dirStateB.Write(t, dirB)
+
+				cwd, err := os.Getwd()
+				if err != nil {
+					t.Fatalf("Failed to get current working directory: %v", err)
+				}
+
+				relA, err := filepath.Rel(cwd, dirA)
+				if err != nil {
+					t.Fatalf("Failed to convert %q to relative path: %v", dirA, err)
+				}
+
+				return relA, dirB
+			},
+			checkResult: checkSuccess,
+		},
+		{
+			name: "success with path B relative",
+			setup: func(t *testing.T, tmpDir string) (string, string) {
+				dirA := filepath.Join(tmpDir, "a")
+				dirB := filepath.Join(tmpDir, "b")
+
+				dirStateA.Write(t, dirA)
+				dirStateB.Write(t, dirB)
+
+				cwd, err := os.Getwd()
+				if err != nil {
+					t.Fatalf("Failed to get current working directory: %v", err)
+				}
+
+				relB, err := filepath.Rel(cwd, dirB)
+				if err != nil {
+					t.Fatalf("Failed to convert %q to relative path: %v", dirB, err)
+				}
+
+				return dirA, relB
+			},
+			checkResult: checkSuccess,
+		},
+		{
+			name: "error with directory A does not exist",
+			setup: func(t *testing.T, tmpDir string) (string, string) {
+				dirA := filepath.Join(tmpDir, "a")
+				dirB := filepath.Join(tmpDir, "b")
+
+				if err := os.Mkdir(dirB, 0755); err != nil {
+					t.Fatalf("Failed to create directory %q: %v", dirB, err)
+				}
+
+				return dirA, dirB
+			},
+			checkResult: func(t *testing.T, dirA, dirB string, err error) {
+				if !os.IsNotExist(err) {
+					t.Errorf("Expected a directory not exists error, got %v", err)
+				}
+			},
+		},
+		{
+			name: "error with directory B does not exist",
+			setup: func(t *testing.T, tmpDir string) (string, string) {
+				dirA := filepath.Join(tmpDir, "a")
+				dirB := filepath.Join(tmpDir, "b")
+
+				if err := os.Mkdir(dirA, 0755); err != nil {
+					t.Fatalf("Failed to create directory %q: %v", dirA, err)
+				}
+
+				return dirA, dirB
+			},
+			checkResult: func(t *testing.T, dirA, dirB string, err error) {
+				if !os.IsNotExist(err) {
+					t.Errorf("Expected a directory not exists error, got %v", err)
+				}
+			},
+		},
```

Comment:
should we add a case where:
- one dir is empty
- or even both dirs are empty 

?